src/tsolvers/egraph/EgraphSolver.cc

```diff
@@ -1575,3 +1596,14 @@ void Egraph::printStatistics(std::ostream & os) {
     TSolver::printStatistics(os);
     egraphStats.printStatistics(os);
 }
+
+void Egraph::reanalyze(ERef eref) {
+    // FIXME: This will probably never be false, we need to find a different condition
```

Comment:
Could it simply be `backtrack_points.size() > 1` ?  I also think that this has to be always true, since the backtrack point in `TSolver` is popped only after the call to `Egraph::backtrackToStackSize()` returns.  I don't have test cases that I could quickly use for checking the theory though.  If you don't have either, maybe we can postpone the discussion to once MBTC is available.

Reply:
That would probably work, but I still think this can stop much sooner.

I agree that this discussion and analysis will be much easier when you can try some real examples and see how this behaves.

Reply:
> That would probably work, but I still think this can stop much sooner.

Maybe so.  In any case let's keep the `FIXME` there for now obviously.